src/link/MachO/Dylib.zig

```diff
@@ -71,9 +72,53 @@ pub fn closeFile(self: Dylib) void {
     }
 }
 
+pub fn decodeArch(cputype : macho.cpu_type_t) !std.Target.Cpu.Arch {
+    const arch: std.Target.Cpu.Arch = switch (cputype) {
+        macho.CPU_TYPE_ARM64 => .aarch64,
+        macho.CPU_TYPE_X86_64 => .x86_64,
+        else => |value| {
+            log.err("unsupported cpu architecture 0x{x}", .{value});
+            return error.UnsupportedCpuArchitecture;
+        },
+    };
+    return arch;
+}
+
 pub fn parse(self: *Dylib) !void {
     log.debug("parsing shared library '{s}'", .{self.name.?});
 
+    var library_offset : u64 = 0;
+    {
+        const fat_header = try self.file.?.reader().readStructEndian(macho.fat_header, builtin.Endian.Big);
+        if(fat_header.magic == macho.FAT_MAGIC) {
+            var fat_arch_index: u32 = 0;
+            var found_arch = false;
+            while (fat_arch_index < fat_header.nfat_arch) : (fat_arch_index += 1) {
+                const fat_arch = try self.file.?.reader().readStructEndian(macho.fat_arch, builtin.Endian.Big);
+                // If we come across an architecture that we do not know how to handle, that's
+                // fine because we can keep looking for one that might match.
+                const libArch = decodeArch(fat_arch.cputype) catch |err| switch(err) {
```

Comment:
RE: naming conventions. Variables are written in `snake_case`, methods in `camelCase`, types start with a capital letter and constants are (usually) in all-caps.

Reply:
Yes cheers! This was just out of habit from - will make sure to sort these!